mesonbuild/interpreterbase.py

```diff
@@ -31,6 +31,15 @@ def check_stringlist(a, msg='Arguments must be strings.'):
         mlog.debug('Element not a string:', str(a))
         raise InvalidArguments(msg)
 
+def check_typeof(a, types, msg=None):
+    if not isinstance(a, list):
+        mlog.debug('Not a list:', str(a))
+        raise InvalidArguments('Argument not a list.')
+    if not all(any([isinstance(s, t) for t in types]) for s in a):
+        mlog.debug('Element not {}: {}'.format(" or ".join((t.__name__ for t in types)), str(a)))
+        raise InvalidArguments(msg if not msg is None else
+                'Arguments must be {}.'.format(" or ".join((t.__name__ for t in types))))
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->